src/Internal/Posix/PosixRunner.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+namespace Amp\Process\Internal\Posix;
+
+use Amp\ByteStream\ReadableResourceStream;
+use Amp\ByteStream\WritableResourceStream;
+use Amp\Process\Internal\ProcessHandle;
+use Amp\Process\Internal\ProcessRunner;
+use Amp\Process\Internal\ProcessStatus;
+use Amp\Process\ProcessException;
+use Revolt\EventLoop;
+
+final class PosixRunner implements ProcessRunner
+{
+    private const FD_SPEC = [
+        ["pipe", "r"], // stdin
+        ["pipe", "w"], // stdout
+        ["pipe", "w"], // stderr
+        ["pipe", "w"], // exit code pipe
+    ];
+
+    /** @var string|null */
+    private static ?string $fdPath = null;
+
+    public function start(
+        string $command,
+        string $workingDirectory = null,
+        array $environment = [],
+        array $options = []
+    ): ProcessHandle {
+        if (!\extension_loaded('pcntl')) {
+            throw new ProcessException('Missing ext-pcntl to run processes with PosixRunner');
+        }
+
+        if (!\extension_loaded('posix')) {
+            throw new ProcessException('Missing ext-posix to run processes with PosixRunner');
+        }
+
+        $command = \sprintf(
+            '{ (%s) <&3 3<&- 3>/dev/null & } 3<&0; trap "" INT TERM QUIT HUP;' .
+            'pid=$!; echo $pid >&3; wait $pid; RC=$?; echo $RC >&3; exit $RC',
+            $command
+        );
+
+        $proc = @\proc_open(
```

Comment:
Should we replace any uses of `@` to set and restore an error handler?

Reply:
Might be a good idea, yes.